### PR TITLE
Refactor version constraints

### PR DESCRIPTION
**Pull request by aeisenberg**
A simple refactoring that simplifies and unifies how we check if a
feature is supported by a specific cli version.

This is an internal change and not visible to end users.

## Checklist

- [n/a] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [n/a] Issues have been created for any UI or other user-facing changes made by this pull request.
- [n/a] `@github/docs-content-codeql` has been cc'd in all issues for UI or other user-facing changes made by this pull request.

### CODE REVIEW

File: extensions/ql-vscode/src/cli.ts [hunk 1]
```diff
@@ -1030,3 +996,56 @@ export function shouldDebugQueryServer() {
...
+   * CLI version where database registration was introduced
+   */
+  public static CLI_VERSION_WITH_DB_REGISTRATION = new SemVer('2.4.1');
+
+  constructor(private readonly cli: CodeQLCliServer) {
+    /**/
+  }
+
+  public async supportsDecompileDil() {
+    return (await this.cli.getVersion()).compare(CliVersionConstraint.CLI_VERSION_WITH_DECOMPILE_KIND_DIL) >= 0;
```
Comment (adityasharad) [inline extensions/ql-vscode/src/cli.ts:1033]: You could factor out this pattern into a helper `async isVersionAtLeast(v: SemVer): boolean`

File: extensions/ql-vscode/src/cli.ts [hunk 2]
```diff
@@ -1030,3 +996,60 @@ export function shouldDebugQueryServer() {
...
+  public async supportsDowngrades() {
+    return this.isVersionAtLeast(CliVersionConstraint.CLI_VERSION_WITH_DOWNGRADES);
+  }
+
+  public async supportsResolveQlref() {
+    return this.isVersionAtLeast(CliVersionConstraint.CLI_VERSION_WITH_RESOLVE_QLREF);
+  }
+
+  async supportsDatabaseRegistration() {
+    return (await this.cli.getVersion()).compare(CliVersionConstraint.CLI_VERSION_WITH_DB_REGISTRATION) >= 0;
```
Comment (edoardopirovano) [inline extensions/ql-vscode/src/cli.ts:1053]: Shouldn't this also be using the new `isVersionAtLeast`  method?

Comment (aeisenberg) [reply extensions/ql-vscode/src/cli.ts:1053]: Yep. Missed that one. Thanks.

**Review by edoardopirovano — APPROVED** [review]
Looks good to me!

**Pull request merged**